### DISCUSSION TITLE
Ideas from @rogersad

### DISCUSSION THREAD

**Discussion by broofa** [broofa/flightcard#6] [General]
@rogersad wrote:

----
I've written apps in the past. Simplest way to do that is use google forms. That requires internet access though. I don't have any connection at Brothers. I think this is the biggest problem.
Other option is write an actual app/smart page, but that usually requires net connection also in order to download/submit the data, and somebody has to host the db server.
ideas:
shared tablet with cel service that everybody uses (1 or more of these), or use your own phone/tablet if you have connection - would i require a log in? or you take my word I am who I say I am?
rso on duty would have to be signed in to the system to do approvals - and not forget to sign out when the shift is over. 
as for the rest, the db should have all members names with nar/tra numbers and level, just pick your name from the list and you get pre-filled.
date and time all get auto-filled when rso approves. 
maybe have pick list of standard motors? but thats a loooong list. problem of NOT doing this is somebody has to add up all those newtons, or it gets done on the back-end reporting page. Could be done by just picking the A/B/C/... and filling in the rest.
bonus points if you pull in current weather conditions (wind/temp) from some web service

edit: Been so long I forgot... a local app, with local storage, you wouldn't need the net connection, but you'd need a way to sync up multiple clients.

**Comment by broofa** [comment]
Hi Roger, thanks for the suggestions.  You're spot on with these issues.  We should probably move some/all of these to separate threads if conversation continues but, briefly...

> That requires internet access though. I don't have any connection at Brothers. 

Yes, this is one of the bigger issues.  I believe there is *some* data availability at Brothers.  E.g. I get 3 bars with Verizon out there.  Obviously not having everyone connected is less than ideal, but for the initial version I'm okay asking people to setup ad-hoc hotspots.  The app itself should be very data "light".  The production webapp currently weighs in at < 1MB of HTML+JS+CSS.  It uses Firebase (realtime database) that is also very lean.  Most user actions and state changes will transfer a few hundred bytes of data.

> would i require a log in

Yes.  Moreover I don't intend to offer anonymous or "guest" logins.  Users will login via their Facebook, Google, Apple, or Github account.  (Currently only Google accounts are supported.)  I.e. Users will be expected to use their real identities on the app.

> rso on duty would have to be signed in to the system to do approvals 

Yes.  Users can be granted "officer" permissions.  Officers can select what "role" (LCO or RSO) they are currently on duty for, which will be visible to all users.

> the db should have all members names with nar/tra numbers and level

Yes.  You enter this information when you "check in" to a launch.  Your certification level will be visible to all users, but until an officer verifies your certification (i.e. you show them your cert card), it will appear with a "⚠️" symbol next to it.

> maybe have pick list of standard motors

thrustcurve.org has [a nice API](https://www.thrustcurve.org/info/api.html) for this.   I've actually got a [motor search field component](https://github.com/broofa/flightcard/blob/ae5c8d2398dd76be2b60fbd82a25c3f6600ff395/components/MotorQuery.tsx) that provides type-ahead suggestions.  E.g. type "J350" and it'll suggest any motors that mach that "common name".  I haven't integrated it yet because I need to figure out how best to account for the much richer set of data you end up getting with this information.

FWIW,  I actually see this as one of the more compelling use cases for this app.  Better/more detailed motor information enables some pretty cool capabilities.  E.g. Automatic thrust:weight ratio calculations, rail velocity estimates, etc.

> bonus points if you pull in current weather conditions (wind/temp) from some web service

Yup. Lots of good weather APIs available to choose from.  Note, too, that location also gives you ground elevation, which gives you a starting point or air pressure/density calculations so with that, motor thrust data, and rocket diameter/weight/length, it's possible to do a [crude] flight simulation profile.

**Comment by rogersad** [comment]
i'm still thinking about no service for most people. This brings me back to the shared tablet idea.
you have 2or 3 tablets with service. i pick my name from the  list and it just prompts me for a 4-6 digit passcode and I can proceed to enter data, then pass the tablet to the next guy in line. The RSO/LCO would have an aggregate view which would show everything currently approved/needs approved, on pad, etc.

i guess you'd have several status: new (pending rso), approved/denied, ready for launch (on pad), success/fail

also, anybody who's used open rocket knows that thrustcurve list can be pretty daunting without a filter. maybe at least select manufacture and impulse first, then it would be a short list to select the designation.

**Reply by broofa** [reply]
> thinking about no service for most people

This isn't something I'm particularly interested in trying to solve.  At least not for v1 of this app.

While I agree this will be an issue, building an offline version of the app is a completely different undertaking than what I'm currently envisioning.  So, for now, I'm okay with saying that an internet connection is a requirement.

> thrustcurve list can be pretty daunting

Yes and no.  There's only ~1K motors in the thrustcurve database.  Entering the "commonName" (size+thrust ala "J350") typically narrows it down to 3-4 motors. But, yes, whatever the solution is would have to avoid overwhelming the user with options.
